src/einsteinpy/geodesic/utils.py

```diff
@@ -0,0 +1,231 @@
+"""
+Utilities for Geodesic Module
+
+Unit System: M-Units => G = c = M = 1
+
+"""
+import warnings
+
+import numpy as np
+
+
+def _energy(E, q, p, a, mu):
+    """
+    Utility function to compute Energy of the test particle,
+    using `scipy.optimize.fsolve`
+
+    Parameters
+    ----------
+    E : float
+        Variable, which `scipy.optimize.fsolve` solves for
+    q : array_like
+        Length-3 Array, containing the initial 3-Position
+    p: array_like
+        Length-3 Array, containing the initial 3-Momentum
+    a : float
+        Dimensionless Spin Parameter of the Black Hole
+        ``0 <= a <= 1``
+    mu : float
+        Rest Mass of the test particle
+        ``mu = 0.`` for Null Geodesics
+        ``mu = -1`` for Timelike Geodesics
+
+    Returns
+    -------
+    float
+        Energy of the test particle
+
+    """
+    return (
+        a ** 4 * (-(E ** 2) + mu ** 2 + 2 * p[0] ** 2)
+        + 8 * a * E * p[2] * q[0]
+        - a ** 2
+        * (
+            2 * p[2] ** 2
+            - 2 * p[1] ** 2
+            + q[0]
+            * (
+                mu ** 2 * (2 - 3 * q[0])
+                - 4 * p[0] ** 2 * (-2 + q[0])
+                + E ** 2 * (2 + 3 * q[0])
+            )
+        )
+        + 2
+        * q[0]
+        * (
+            p[1] ** 2 * (-2 + q[0])
+            + q[0]
+            * (
+                p[0] ** 2 * (-2 + q[0]) ** 2
+                + q[0] * (mu ** 2 * (-2 + q[0]) - E ** 2 * q[0])
+            )
+        )
+        - (a ** 2 + (-2 + q[0]) * q[0])
+        * (
+            a ** 2 * (E - mu) * (E + mu) * np.cos(2 * q[1])
+            - 2 * p[2] ** 2 * (1 / (np.sin(q[1]) ** 2))
+        )
+    ) / (
+        (a ** 2 + (-2 + q[0]) * q[0])
+        * (a ** 2 + 2 * q[0] ** 2 + a ** 2 * np.cos(2 * q[1]))
+    )
+
+
+def _sphToCart(r, th, ph):
+    """
+    Utility function to convert Spherical Polar
+    Coordinates to Cartesian Coordinates
+
```

Comment:
Could you add `Parameters` and `Returns` to the docstring?

Reply:
Sure. Done.